account_invoice_validate_send_email/models/account_invoice.py

```diff
@@ -109,3 +113,14 @@ def action_send(self):
                 )
                 self.invoice_sent = True
         return True
+
+    # This method not supposed red_order. Because red_order haven't delivery of own company.
+    @api.multi
+    def _compute_slip_number(self):
+        if self.type not in ("out_refund"):
+            for invoice in self:
+                tracking_numbers = ""
+                for picking in invoice.picking_ids:
+                    if picking.carrier_tracking_ref:
+                        tracking_numbers += picking.carrier_tracking_ref + "\n"
+                invoice.slip_number = tracking_numbers
```

Comment:
```suggestion
    def _compute_carrier_tracking_refs(self):
        for invoice in self:
            invoice.carrier_tracking_refs = ", ".join(
                x.carrier_tracking_ref for x in invoice.picking_ids if x.carrier_tracking_ref
            )
```

A few points to note:

- `if self.type not in ("out_refund"):` would result in a server error when you work on multiple invoices.
- There is no point adding the condition by invoice type - the return case is naturally filtered out by not having linked pickings.
- Regarding `tracking_numbers += picking.carrier_tracking_ref + "\n"`, it's not ideal that the value ends with a new line.
- Pay attention to the model attribute order described in https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#models.

Reply:
Thank you.
I understand.